File: puppeteer-python/fullteaching_e2e/utils/parameter_loader.py
# -*- coding: utf-8 -*-
"""pytest.mark.parametrize data providers, ported from utils/ParameterLoader.java.

Java exposes these as `@MethodSource` streams; the pytest idiom is a plain list of
tuples consumed directly by `@pytest.mark.parametrize("mail,password,role", ...)`.
"""
import logging
from typing import List, Tuple

from fullteaching_e2e.utils.user_loader import get_all_users

logger = logging.getLogger(__name__)

UserParams = List[Tuple[str, str, str]]


def _is_student(user) -> bool:
    return user.role.strip().upper() == "STUDENT"


def _is_teacher(user) -> bool:
    return user.role.strip().upper() == "TEACHER"


def get_test_users() -> UserParams:
    users = get_all_users()
    logger.debug("get_test_users -- %d users", len(users))
    return [(u.name, u.password, u.role) for u in users]


def get_test_students() -> UserParams:
    users = get_all_users()
    return [(u.name, u.password, u.role) for u in users if _is_student(u) and not _is_teacher(u)]


def get_test_teachers() -> UserParams:
    users = get_all_users()
    return [(u.name, u.password, u.role) for u in users if _is_teacher(u) and not _is_student(u)]

File: playwright-csharp/Utils/ParameterLoader.cs
using NUnit.Framework;

namespace FullTeaching.E2E.Utils;

/// <summary>
/// NUnit [TestCaseSource] data providers, ported from utils/ParameterLoader.java. NUnit's
/// TestCaseSource is the closest 1:1 match of any of the sibling ports to JUnit's
/// @ParameterizedTest + @MethodSource - same "static method returning test data" shape.
/// </summary>
public static class ParameterLoader
{
    public static IEnumerable<TestCaseData> GetTestUsers() =>
        UserLoader.GetAllUsers().Select(ToTestCase);

    public static IEnumerable<TestCaseData> GetTestStudents() =>
        UserLoader.GetAllUsers().Where(u => IsStudent(u) && !IsTeacher(u)).Select(ToTestCase);

    public static IEnumerable<TestCaseData> GetTestTeachers() =>
        UserLoader.GetAllUsers().Where(u => IsTeacher(u) && !IsStudent(u)).Select(ToTestCase);

    private static bool IsStudent(User user) => user.Role.Trim().Equals("STUDENT", StringComparison.OrdinalIgnoreCase);

    private static bool IsTeacher(User user) => user.Role.Trim().Equals("TEACHER", StringComparison.OrdinalIgnoreCase);

    private static TestCaseData ToTestCase(User user) =>
        new TestCaseData(user.Name, user.Password, user.Role).SetName($"{{m}}({user.Role}_{user.Name})");
}
